Add CISA Secure by Design Cheat Sheet

Prasad-JB opened the pull request:
This PR adds a new cheat sheet based on CISA’s Secure by Design principles.

- New file: CISA_Secure_By_Design_Cheat_Sheet.md
- Organized around the 7 CISA principles
- Includes practical, developer-friendly guidance under each heading
- References link to the official CISA alert and PDF

This should give teams a dedicated resource for applying the Secure by Design principles, while complementing the existing Secure Product Design cheat sheet.

Looking forward to feedback from the community

szh commented on cheatsheets/CISA_Secure_By_Design_Cheat_Sheet.md:
```diff
@@ -0,0 +1,123 @@
+---
+layout: col-sidebar
+title: CISA Secure by Design Cheat Sheet
+tags: CISA, Secure-by-Design, Development, Principles
+level: 2
+type: cheatsheet
+pitch: Practical guidance for implementing CISA's Secure by Design principles in software development.
+---
```
We don't use this on any other of the cheat sheets. What's the purpose of it?

szh commented on cheatsheets/CISA_Secure_By_Design_Cheat_Sheet.md:
```diff
@@ -0,0 +1,123 @@
+- **Cross-functional Teams**: Create teams with both development and security expertise
+- **Security Training**: Provide ongoing security education for all developers
+
+### 4. Secure the Software Development Lifecycle (SDLC)
```
I only see the first 3 principles in the [whitepaper](https://www.cisa.gov/sites/default/files/2023-10/SecureByDesign_1025_508c.pdf). Where are 4-7 from?

szh commented on cheatsheets/CISA_Secure_By_Design_Cheat_Sheet.md:
```diff
@@ -0,0 +1,123 @@
+## Tools and Resources
+
+- **CISA Secure by Design Alert**: [AA23-074A](https://www.cisa.gov/news-events/cybersecurity-advisories/aa23-074a)
+- **CISA Secure by Design Principles**: [Full PDF Guide](https://www.cisa.gov/sites/default/files/2023-04/secure_by_design_alert_4.19.23.pdf)
```
I'm getting a 404 for this. Did you mean this?
```suggestion
- **CISA Secure by Design Principles**: [Full PDF Guide](https://www.cisa.gov/sites/default/files/2023-10/SecureByDesign_1025_508c.pdf)
```

szh commented on cheatsheets/CISA_Secure_By_Design_Cheat_Sheet.md:
```diff
@@ -0,0 +1,123 @@
+
+- **CISA Secure by Design Alert**: [AA23-074A](https://www.cisa.gov/news-events/cybersecurity-advisories/aa23-074a)
+- **CISA Secure by Design Principles**: [Full PDF Guide](https://www.cisa.gov/sites/default/files/2023-04/secure_by_design_alert_4.19.23.pdf)
+- **OWASP Secure Product Design Cheat Sheet**: [Link to related cheat sheet]
```
```suggestion
- [OWASP Secure Product Design Cheat Sheet](Secure_Product_Design_Cheat_Sheet.md)
```

szh commented on cheatsheets/CISA_Secure_By_Design_Cheat_Sheet.md:
```diff
@@ -0,0 +1,123 @@
+- **CISA Secure by Design Alert**: [AA23-074A](https://www.cisa.gov/news-events/cybersecurity-advisories/aa23-074a)
+- **CISA Secure by Design Principles**: [Full PDF Guide](https://www.cisa.gov/sites/default/files/2023-04/secure_by_design_alert_4.19.23.pdf)
+- **OWASP Secure Product Design Cheat Sheet**: [Link to related cheat sheet]
+- **Memory Safety**: [CISA Memory Safety Roadmap](https://www.cisa.gov/resources-tools/resources/memory-safety-roadmap)
```
This gives me a 404 as well

szh commented on cheatsheets/CISA_Secure_By_Design_Cheat_Sheet.md:
```diff
@@ -0,0 +1,123 @@
+
+1. [CISA Secure by Design](https://www.cisa.gov/securebydesign)
+2. [CISA Secure by Design Alert AA23-074A](https://www.cisa.gov/news-events/cybersecurity-advisories/aa23-074a)
+3. [CISA Memory Safety Recommendations](https://www.cisa.gov/resources-tools/resources/memory-safety-roadmap)
```
Same link, so 404 here too

szh commented on cheatsheets/CISA_Secure_By_Design_Cheat_Sheet.md:
```diff
@@ -0,0 +1,123 @@
+
+- Prasad-JB
+- OWASP Cheat Sheets Team
+- CISA Cybersecurity Division
```
I'm not sure if we can put this here if they don't directly contribute to this cheat sheet. It wouldn't be fair to them to claim they're involved if they aren't.

szh reviewed (changes requested):
The first three principles are great. I left some questions about the rest.

Prasad-JB commented:
Hi @szh and @mackowski,

Thanks for the feedback I’ve cleaned up the PR:
Removed the new CISA cheat sheet.

- No ASVS changes.
- Added just a few lightweight CISA points to the existing Secure Product Design Cheat Sheet (cross-functional teams, security training, memory safety).
-All links work and lint checks pass.

**Should be ready for review now**
Prasad JB
